charmcraft.yaml

```diff
@@ -169,6 +169,13 @@ config:
       description: |
         The GitHub runner version to use, e.g. 2.317.0. Empty default value will fetch the latest
         version by default. See https://github.com/actions/runner/releases.
+    script-url:
+      type: string
+      default: ""
+      description: |
+        URL to external script to run during cloud-init. Allows user-defined customization scripts
+        to be run. This script is run as root, within the cloud-init user-data script.
```

Comment:
Did we also need secrets to be passed to the script? Or are you thinking of doing that in a follow up PR? A follow up would probably be better

Reply:
Yup this would be a separate follow up PR to make this one small and targeted for review :D